src/plugins/mf_priority.cpp

```diff
@@ -331,6 +331,25 @@ static json_t *user_to_json (
 }
 
 
+// Scan the users map and look at each user's default bank to see if any one
+// of them have a valid bank (i.e one that is not "DNE"; if any of the users do
+// do have a valid bank, it will return false)
+static bool check_map_for_dne_only ()
```

Comment:
A really quick thought here before a full review.

If this function instead returned the default bank for a user or DNE if the user has no default bank, would it potentially be useful elsewhere? Also, it might be more clear if the users map was passed as a parameter to the function...

Reply:
I think there are a couple of other places where the default bank for a user is queried, but I had actually posted an experimental PR (#371) where the bank lookup was moved to its own helper function. I had originally thought that it would be useful to have functionality like the one proposed in this PR (just scanning the users map for the first non-DNE bank it sees) kept separate, but perhaps it would be better to combine the two? Off the top of my head, maybe in the check where it looks for a default bank, it could add something like:

```c++
else {
        bank = const_cast<char*> (users_def_bank[userid].c_str ());
        bank_it = it->second.find (std::string (bank));
+       if (it->second == "DNE")
+           return {DNE_DEFAULT_BANK, bank_it};
        else if (bank_it == it->second.end ())
            return {NO_DEFAULT_BANK, bank_it};
    }
```

> Also, it might be more clear if the users map was passed as a parameter to the function...

The `users` map is a global variable in the plugin. Would it be sufficient to add a comment in this function to mention this?

Reply:
Yeah, a comment would definitely help!

Reply:
just force-pushed up a change to add a comment about both the `users` and `users_def_bank` maps since they are both global variables.